app/scripts/components/common/dateslider/data-points.tsx

```diff
@@ -78,15 +78,15 @@ export function DataPoints(props: DataPointsProps) {
     const rootG = select(container.current);
 
     const classAttr = (d, c) => {
-      return hoveringDataPoint === d.date ? `${c} over` : c;
+      return hoveringDataPoint === d.index ? `${c} over` : c;
```

Comment:
Have you checked whether the performance issue was also caused by this? The whole `useEffect` block gets re-executed everytime the hovered point changes. The points could be rendered "statically" (this `useEffect`would only have a dependency on `data`), with another `useEffect` on top of it, with only a dependency on `hoveringDataPoint`, that only takes care of drawing the single hovered DOM element.

Reply:
The performance problems were particularly prominent on render, not so much on running. Adding all the DOM nodes to show 7000+ data points is what was causing the problems. Once they were in the DOM, it ran smoothly.

It is true that the useEffect would run multiple times, but once the nodes were there, d3 wouldn't do anything else to them.